### DISCUSSION TITLE
LiveWire request url contains the asset url

### DISCUSSION THREAD

**Discussion by LouisGac** [livewire/livewire#3714] [1. Help] [answered]
Hi,

First of all thanks for the good work, livewire is beautiful and a perfect alternative to React or vue . 

 I use livewire in a very exotic environment. At work we are refactoring our legacy application, so Laravel is a subdirectory of Legacy. This implies that the public directory that serves the css and js files is on top of the Laravel folder:

we have legacy/laravel with the laravel code 
and legacy/res/laravel/js for the js files associated to laravel

I manage to load the livewire JS by modifying the configuration:
'asset_url' => '/res/laravel/js'

My problem:
Following the livewire counter tuto, when I click I get a call to:

 https://mywebsite.com/res/laravel/js/livewire/message/counter

instead of a call to:

 https://mywebsite.com/laravel/message/counter

How can I tell livewire.js that its calls should not be made relative to the path of the asset url, but according to a totally different and custom url?

**Comment by denizozturk** [comment]
I don't think this configuration can be done with `config/livewire.php`

By updating the lines below, you will get rid of the `livewire/message` path.
![lw_provider](https://user-images.githubusercontent.com/34531750/133145243-16b4c7e9-4174-40e6-bd54-314f430b803e.png)

![lw_js](https://user-images.githubusercontent.com/34531750/133145240-d661ebce-5d71-4ba3-bc3d-eb60facf6de9.png)
I marked the line in the `index.js` file so that you can see and understand the relevant field. You must make the change in the `livewire.js` file.

**Comment by LouisGac** [comment, marked as answer]
Thank you very much, you helped me a lot.
Indeed, I was not trying to change "/livewire/message/", but window.livewire_app_url.
So I've just defined window.livewire_app_url in a script tag in the footer and everything works like a charm. 
I've also added a route for /laravel/livewire/message
